src/nanopolish_index.cpp

```diff
@@ -326,44 +343,71 @@ int index_main(int argc, char** argv)
-    process_summary_fofn();
-    std::multimap<std::string, std::string> fast5_to_read_name_map;
-    for(const auto& ss_filename : opt::sequencing_summary_files) {
-        if(opt::verbose > 2) {
-            fprintf(stderr, "summary: %s\n", ss_filename.c_str());
+    if(opt::slow5file.empty()){
+        // Read a map from fast5 files to read name from the sequencing summary files (if any)
+        process_summary_fofn();
+        std::multimap<std::string, std::string> fast5_to_read_name_map;
+        for(const auto& ss_filename : opt::sequencing_summary_files) {
+            if(opt::verbose > 2) {
+                fprintf(stderr, "summary: %s\n", ss_filename.c_str());
+            }
+            parse_sequencing_summary(ss_filename, fast5_to_read_name_map);
         }
-        parse_sequencing_summary(ss_filename, fast5_to_read_name_map);
-    }
 
-    // Detect non-unique fast5 file names in the summary file
-    // This occurs when a file with the same name (abc_0.fast5) appears in both fast5_pass
-    // and fast5_fail. This will be fixed by ONT but in the meantime we check for
-    // fast5 files that have a non-standard number of reads (>4000) and remove them
-    // from the map. These fast5s will be indexed the slow way.
-    clean_fast5_map(fast5_to_read_name_map);
+        // Detect non-unique fast5 file names in the summary file
+        // This occurs when a file with the same name (abc_0.fast5) appears in both fast5_pass
+        // and fast5_fail. This will be fixed by ONT but in the meantime we check for
+        // fast5 files that have a non-standard number of reads (>4000) and remove them
+        // from the map. These fast5s will be indexed the slow way.
+        clean_fast5_map(fast5_to_read_name_map);
 
-    // import read names, and possibly fast5 paths, from the fasta/fastq file
-    ReadDB read_db;
-    read_db.build(opt::reads_file);
-    bool all_reads_have_paths = read_db.check_signal_paths();
+        // import read names, and possibly fast5 paths, from the fasta/fastq file
+        ReadDB read_db;
+        read_db.build(opt::reads_file);
+        bool all_reads_have_paths = read_db.check_signal_paths();
 
-    // if the input fastq did not contain a complete set of paths
-    // use the fofn/directory provided to augment the index
-    if(!all_reads_have_paths) {
+        // if the input fastq did not contain a complete set of paths
+        // use the fofn/directory provided to augment the index
+        if(!all_reads_have_paths) {
 
-        for(const auto& dir_name : opt::raw_file_directories) {
-            index_path(read_db, dir_name, fast5_to_read_name_map);
+            for(const auto& dir_name : opt::raw_file_directories) {
+                index_path(read_db, dir_name, fast5_to_read_name_map);
+            }
+        }
+
+        size_t num_with_path = read_db.get_num_reads_with_path();
+        size_t num_reads = read_db.get_num_reads();
+        if(num_with_path == 0) {
```

Comment:
This is something modified in f5c where the original nanopolish code  has
```
if(num_with_path == 0) {
        fprintf(stderr, "Error: no fast5 files found\n");
        exit(EXIT_FAILURE);
    } else {
        read_db.print_stats();
        read_db.save();
    }
```    
What is @jts's thought on it?

Reply:
Looks good to me